### PR TITLE
JHub Apps: Filter conda envs by user

### PR DESCRIPTION
**Pull request by aktech**
## Reference Issues or PRs

At the moment it returns all the environments even if the user doesn't have access to it.
Ref: https://github.com/nebari-dev/jhub-apps/issues/51

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [ ] Did you add new tests?

### CODE REVIEW

File: src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub/files/jupyterhub/02-spawner.py [hunk 1]
```diff
@@ -63,6 +65,7 @@ def get_conda_store_environments(query_package: str = ""):
         "/usr/local/etc/jupyterhub/jupyterhub_config.py"
     )
     c.JAppsConfig.hub_host = "hub"
+    c.JAppsConfig.service_workers = 4
```
Comment (dcmcand) [inline src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub/files/jupyterhub/02-spawner.py:68]: @aktech What is going on with this line? It seems unrelated to the issue.

**Review by dcmcand — CHANGES_REQUESTED** [review]
This only allows users to use environments from their personal namespace, not from any shared namespace that they have access to. We need to give them access to any env that they have access to imo.

File: src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub/files/jupyterhub/02-spawner.py (hunk 1 above)
Comment (aktech) [reply src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub/files/jupyterhub/02-spawner.py:68]: Yep, this is unrelated. This is harnessing the new feature being able to configure workers for jhub-apps service.  I thought I would add all jhub apps related settings in this PR, but happy to create an new PR for this.
